### PR TITLE
[Paper submission code - LR & GBT experiments]

### PR DESCRIPTION
**Pull request by rfbr**
## What?
Add LR and GBT solutions to the AdKDD challenge. Also include various experiments to benchmark robustness of these solutions against noise and number of data points.

## Why?
Support the release of the paper, accepted at the WebConf.

## How?
Implementations of the different preprocessing methods to encode the aggregated features + model training and validation.

### CODE REVIEW

File: gbt/main_clicks.ipynb [hunk 1]
```diff
@@ -324,8 +324,8 @@
    "id": "3789bcfb",
    "metadata": {
     "ExecuteTime": {
-     "end_time": "2021-11-19T22:09:00.172234Z",
-     "start_time": "2021-11-19T22:09:00.168240Z"
+     "end_time": "2022-01-14T17:53:52.639607Z",
+     "start_time": "2022-01-14T17:53:42.891Z"
```
Comment (oddskool) [inline gbt/main_clicks.ipynb]: can we avoid committing notebooks with outputs ?

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,30 @@
+autopep8==1.6.0
+black==21.12b0
+click==8.0.3
+cycler==0.11.0
+dataclasses==0.8
+importlib-metadata==4.8.3
```
Comment (oddskool) [inline requirements.txt]: are we sure we need exactly all these libs ?

File: data/fetch_datasets.ipynb [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!wget \"http://go.criteo.net/criteo-ppml-challenge-adkdd21-dataset.zip\"\n",
+    "!unzip criteo-ppml-challenge-adkdd21-dataset.zip && rm criteo-ppml-challenge-adkdd21-dataset.zip\n",
```
Comment (oddskool) [inline data/fetch_datasets.ipynb:17]: is it necessary to unzip all csv files ?
if using numpy it's straightforward to read gzipped files
(not a big deal though)

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+joblib==1.1.0
+kiwisolver==1.3.1
```
Comment (oddskool) [inline requirements.txt]: same: make sure we limit to the exact set of libs required

**Review by oddskool — CHANGES_REQUESTED** [review]

File: gbt/main_clicks.ipynb (hunk 1 above)
Comment (rfbr) [reply gbt/main_clicks.ipynb]: Cleaned in commit [8ed9799](https://github.com/criteo-research/criteo-privacy-preserving-ml-competition/pull/1/commits/8ed979970e0ce0acd03cc5782b109a86af69f0a0).

File: requirements.txt (hunk 2 above)
Comment (rfbr) [reply requirements.txt]: I used a new venv before installing required python packages. I will check again to make sure there is only necessary libs.

File: data/fetch_datasets.ipynb (hunk 3 above)
Comment (rfbr) [reply data/fetch_datasets.ipynb:17]: No it's not, I thought it was cleaner to do it this way. Should we keep .csv.gz files to reduce file space consumption?

File: requirements.txt (hunk 2 above)
Comment (rfbr) [reply requirements.txt]: Updated requirements.txt in commit [8292dec](https://github.com/criteo-research/criteo-privacy-preserving-ml-competition/pull/1/commits/8292decb0bd1180f331e29900c6644470bcfd3d2).

File: requirements.txt (hunk 4 above)
Comment (rfbr) [reply requirements.txt]: Updated requirements.txt in commit [8292dec](https://github.com/criteo-research/criteo-privacy-preserving-ml-competition/pull/1/commits/8292decb0bd1180f331e29900c6644470bcfd3d2).

**Pull request merged**